python/tk_multi_launchapp/prepare_apps.py

```diff
@@ -32,66 +32,95 @@ def prepare_launch_for_engine(engine_name, app_path, app_args, context, file_to_
-    # pass down the file to open into the startup script via env var.
-    if file_to_open:
-        os.environ["TANK_FILE_TO_OPEN"] = file_to_open
-        tk_app.log_debug("Setting TANK_FILE_TO_OPEN to '%s'" % file_to_open)
-
-    # serialize the context into an env var
-    os.environ["TANK_CONTEXT"] = sgtk.context.serialize(context)
-    tk_app.log_debug("Setting TANK_CONTEXT to '%r'" % context)
-
-    # Set environment variables used by apps to prep Tank engine
-    os.environ["TANK_ENGINE"] = engine_name
-
-    # Prep any application specific things now that we know we don't
-    # have an engine-specific bootstrap to use.
-    if engine_name == "tk-maya":
-        _prepare_maya_launch()
-    elif engine_name == "tk-softimage":
-        _prepare_softimage_launch()
-    elif engine_name == "tk-motionbuilder":
-        app_args = _prepare_motionbuilder_launch(app_args)
-    elif engine_name == "tk-3dsmax":
-        app_args = _prepare_3dsmax_launch(app_args)
-    elif engine_name == "tk-3dsmaxplus":
-        app_args = _prepare_3dsmaxplus_launch(context, app_args, app_path)
-    elif engine_name == "tk-photoshop":
-        _prepare_photoshop_launch(context)
-    elif engine_name == "tk-houdini":
-        _prepare_houdini_launch(context)
-    elif engine_name == "tk-mari":
-        _prepare_mari_launch(engine_name, context)
-    elif engine_name in ["tk-flame", "tk-flare"]:
-        (app_path, app_args) = _prepare_flame_flare_launch(
-            engine_name, context, app_path, app_args,
+    try:
+        # Try to use the TK engine to perform the necessary preparations
+        # to launch the DCC. If launcher is None, then chances are the
+        # installed version of the specified engine isn't up-to-date.
+        launcher = sgtk.platform.create_engine_launcher(
+            tk_app.sgtk, context, engine_name
         )
-    else:
-        # This should really be the first thing we try, but some of
-        # the engines (like tk-3dsmaxplus, as an example) have bootstrapping
-        # logic that doesn't properly stand alone and still requires
-        # their "prepare" method here in launchapp to be run. As engines
-        # are updated to handle all their own bootstrapping, they can be
-        # pulled out from above and moved into the except block below in
-        # the way that tk-nuke and tk-hiero have.
-        try:
-            (app_path, app_args) = _prepare_generic_launch(
-                tk_app, engine_name, context, app_path, app_args,
+        if launcher:
+            launch_info = launcher.prepare_launch(app_path, app_args, None, file_to_open)
+            app_path = launch_info.path
+            app_args = launch_info.args
+            os.environ.update(launch_info.environment)
+    except AttributeError:
+        # Assuming 'create_engine_launcher' wasn't found in sgtk.platform
+        # Fallback to use default behavior
+        launcher = None
+    except Exception, e:
```

Comment:
what's the use case here? Seems to me we should just let the exception bubble - sorry if i am missing someting!